empress/taxonomy_utils.py

```diff
@@ -0,0 +1,151 @@
+import warnings
+import pandas as pd
+
+
+class TaxonomyError(Exception):
+    pass
+
+
+class TaxonomyWarning(Warning):
+    pass
+
+
+def split_taxonomy(feature_metadata):
+    """ Attempts to find a taxonomy column and split it into taxonomic levels.
+
+    If one of the columns in the feature metadata DataFrame (ignoring case)
+    is named "Taxon" or "Taxonomy", this will return a new DataFrame where
+    the column in question is removed and replaced with S + 1 new columns,
+    where S is the maximum number of semicolons present in any feature's
+    taxonomy.
+
+    Basically, this lets us convert taxonomy annotations to an easier-to-work-
+    with format; instead of having a single column for taxonomy, we'll have one
+    column per taxonomy level. This will make life easier.
+
+    Parameters
+    ----------
+    feature_metadata : pd.DataFrame
+       DataFrame describing feature metadata.
+
+    Returns
+    -------
+    taxsplit_feature_metadata : pd.DataFrame
+        A version of the input feature metadata split as described above. (If
+        none of the columns in the feature metadata were identified as being
+        taxonomy columns, this DataFrame is identical to the input DataFrame.)
+        The new "Level" columns will be placed at the start of the returned
+        DataFrame's columns.
+
+    Raises
+    ------
+    TaxonomyError
+        If any of the following conditions are met:
+            1. Multiple possible "taxonomy columns" are present in the input
+               DataFrame: for example, both a "Taxon" and "Taxonomy" column are
+               present, or both a "taxon" and "Taxon" column are present.
+            2. A taxonomy column is present in the input DataFrame, and there
+               is already at least one column in the DataFrame that starts with
+               "Level".
+    """
+    # The entries in this tuple should only be lowercase
+    VALID_TAXONOMY_COLUMN_NAMES = ("taxon", "taxonomy")
+    lowercase_col_names = [str(c).lower() for c in feature_metadata.columns]
+
+    # See if there is a "taxonomy column", and do some related validation on
+    # column names
+    invalid_level_columns_present = False
+    tax_col_index = None
+    tax_col_name = None
+    for col, i in zip(lowercase_col_names, range(len(lowercase_col_names))):
```

Comment:
Enumerate is the more pythonic way to do this:

```suggestion
    for i, col in enumerate(lowercase_col_names):
```